[i2c, test] Add a temperature and humidity sensor test for the HDC1080

HU90m opened the pull request:
I don't have permission to edit luís' [hdc1080](https://github.com/luismarques/opentitan/tree/hdc1080), so created a new pr. The old pr: https://github.com/lowRISC/opentitan/pull/18670

- Made the [suggested changes](https://github.com/lowRISC/opentitan/issues/18764)
- Rebased (involved adding kDefaultTimeoutMicros)
- Disabled `kDifI2cSpeedFastPlus` see https://github.com/lowRISC/opentitan/issues/18764

engdoreis commented on sw/device/tests/BUILD:
```diff
@@ -1804,6 +1804,29 @@ opentitan_functest(
+    name = "i2c_host_hdc1080_humidity_temp_test",
+    srcs = ["i2c_host_hdc1080_humidity_temp_test.c"],
+    cw310 = cw310_params(
+        tags = ["skip_in_ci"],  # Requires the BoB in CI.
```
```suggestion
        tags = ["manual"],  # Requires the BoB in CI.
```

engdoreis commented on sw/device/tests/i2c_host_hdc1080_humidity_temp_test.c:
```diff
@@ -0,0 +1,141 @@
+  uint8_t reg = kTemperatureReg;
+  uint8_t data[2] = {0};
+  TRY(i2c_testutils_write(&i2c, kDeviceAddr, 1, &reg, true));
+  busy_spin_micros(100000);
```
Drop this delays as we have a retry with timeout inside the read function below.

engdoreis commented on sw/device/tests/i2c_host_hdc1080_humidity_temp_test.c:
```diff
@@ -0,0 +1,141 @@
+  uint8_t reg = kHumidityReg;
+  uint8_t data[2] = {0};
+  TRY(i2c_testutils_write(&i2c, kDeviceAddr, 1, &reg, true));
+  busy_spin_micros(100000);
```
Drop as above.

engdoreis reviewed (commented):
Just a pointed out a few fixes, other than that it LGTM.

HU90m commented:
- Fixed CI error
- removed spin locks
  - had to increase read timeout to match the old spin lock timeout for the test to pass
- replaced skip tag with manual.
